compiler/noirc_evaluator/src/brillig/brillig_ir/procedures/error_with_string.rs

```diff
@@ -0,0 +1,40 @@
+use acvm::{AcirField, acir::brillig::HeapVector};
+
+use super::ProcedureId;
+use crate::{
+    brillig::brillig_ir::{
+        BrilligContext, ReservedRegisters,
+        debug_show::DebugToString,
+        registers::{RegisterAllocator, ScratchSpace},
+    },
+    ssa::ir::instruction::ErrorType,
+};
+
+impl<F: AcirField + DebugToString, Registers: RegisterAllocator> BrilligContext<F, Registers> {
+    /// Reverts with the error selector of the given string
+    /// This procedure is useful to deduplicate generating code for the same selector.
+    pub(crate) fn call_error_with_string_procedure(&mut self, revert_string: String) {
+        self.add_procedure_call_instruction(ProcedureId::ErrorWithString(revert_string));
+    }
+}
+
+pub(super) fn compile_error_with_string_procedure<F: AcirField + DebugToString>(
+    brillig_context: &mut BrilligContext<F, ScratchSpace>,
+    revert_string: String,
```

Comment:
Still says `revert`. I don't think _revert_ was a controversial term; in the context of blockchain contracts, it's quite common. The transaction that initiated this thing should not be committed to the state tree, even if we don't deal with undoing anything within Noir.